### PR TITLE
Add `↑ahead ↓behind` to status bar

### PR DESCRIPTION
**Pull request by sbayley777**
* adds a `Branch` struct to hold references to calculated data to be shown in the UI.
* display the branch name in the status bar
* display the ahead/behind details to status bar

> Note: there is definitely an issue if the folder doesn't have a git repo initialized.

Screen Shot 2022-04-20 at 11 06 20 am

### CODE REVIEW

File: src/git/branch.rs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        .unwrap()
+        .get()
+        .peel(ObjectType::Commit)
+        .unwrap()
+        .id()
+        .clone();
+      Some(id)
+    }
+  }
+}
```
Comment (sbayley777) [inline src/git/branch.rs:103]: I haven't done any specs yet. ill do them after this is reviewed as i may have heavily sinned in this PR.

File: src/ui.rs [hunk 2]
```diff
@@ -34,8 +34,17 @@ pub fn render_left_view<B: Backend>(f: &mut Frame<B>, app: &mut App, area: Rect)
...
-pub fn render_status<B: Backend>(f: &mut Frame<B>, _app: &mut App, area: Rect) {
-  let paragraph = Paragraph::new("This is the current status")
+pub fn render_status<B: Backend>(f: &mut Frame<B>, app: &mut App, area: Rect) {
+  let (ahead, behind) = app.branch.ahead_behind;
+
+  let name = format!(
+    "{}↑ {}↓ -> {}",
+    ahead,
+    behind,
+    app.branch.name.as_ref().unwrap_or(&"Unknown".to_string())
```
Comment (sbayley777) [inline src/ui.rs:44]: i think colourizing the formatting of ahead/behind would be an easy task for someone to pickup after this is merge din

File: src/git/branch.rs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    } else {
+      let id = branch
+        .as_ref()
+        .unwrap()
+        .get()
+        .peel(ObjectType::Commit)
+        .unwrap()
+        .id()
+        .clone();
+      Some(id)
```
Comment (sbayley777) [inline src/git/branch.rs:100]: i've came accross an easier way to get the `Oid` on the weekend

https://github.com/xcambar/purs/blob/master/src/precmd.rs#L70-L83

**Comment by colinbankier** [comment]
Looking very cool. Sorry haven't had time to review it in the last week. Will back to more rustiness after this week! Maybe you can even walk us through it in the rust session?

**Review by colinbankier — APPROVED** [review]
Awesome work!

**Pull request merged**